### PR TITLE
encrypt passwords for files_external

### LINKED ISSUE #21661: Properly encrypt password in oc_external_config
Before migrating from mount.json to the database, passwords were encrypted in the config file.

Now that we are using the database, the same approach must be used.

@icewind1991 @LukasReschke

### PR DESCRIPTION
**Pull request by icewind1991**
Fixes #21661

cc @PVince81 @LukasReschke @Xenopathic

### CODE REVIEW

File: apps/files_external/service/dbconfigservice.php [hunk 1]
```diff
@@ -41,12 +42,18 @@ class DBConfigService {
...
 
 	/**
+	 * @var ICrypto
+	 */
+	private $crypto;
+
+	/**
 	 * DBConfigService constructor.
 	 *
 	 * @param IDBConnection $connection
```
Comment (LukasReschke) [inline apps/files_external/service/dbconfigservice.php]: PHPDoc :smile:

Comment (icewind1991) [inline apps/files_external/service/dbconfigservice.php]: done

**Comment by PVince81** [comment]
I'm slightly worried about people who updated to the beta, will it recognize non-encrypted password ?

**Comment by icewind1991** [comment]
> will it recognize non-encrypted password ?

Yes

**Comment by PVince81** [comment]
Tested, works :+1:

**Comment by rullzer** [comment]
I can still add and mount my samba share. :)
Code looks good. :+1:

**Pull request merged**